apis/pipelines/conversion_utils.go

```diff
@@ -11,41 +12,52 @@ type ConversionRemainder interface {
 	ConversionAnnotation() string
 }
 
-func SetConversionAnnotations(resource metav1.Object, remainders ...ConversionRemainder) error {
+func SetConversionAnnotations(
+	resource metav1.Object,
+	remainders ...ConversionRemainder,
+) error {
 	annotations := resource.GetAnnotations()
-
 	for _, remainder := range remainders {
 		if !remainder.Empty() {
 			remainderJson, err := json.Marshal(remainder)
 			if err != nil {
 				return err
 			}
-
 			if annotations == nil {
 				annotations = map[string]string{}
 			}
 			annotations[remainder.ConversionAnnotation()] = string(remainderJson)
 		}
 	}
-
 	resource.SetAnnotations(annotations)
-
 	return nil
 }
 
-func RetrieveAndUnsetConversionAnnotations(resource metav1.Object, remainders ...ConversionRemainder) error {
+func GetAndUnsetConversionAnnotations(
+	resource metav1.Object,
+	remainders ...ConversionRemainder,
+) error {
 	annotations := resource.GetAnnotations()
-
 	for _, remainder := range remainders {
 		if remainderJson, hasRemainder := annotations[remainder.ConversionAnnotation()]; hasRemainder {
 			err := json.Unmarshal([]byte(remainderJson), remainder)
 			if err != nil {
 				return err
 			}
-
 			delete(annotations, remainder.ConversionAnnotation())
 		}
 	}
+	return nil
+}
 
+func TransformInto[S any, D any](source S, destination *D) error {
+	srcString, err := json.Marshal(source)
+	println(string(srcString))
```

Comment:
```suggestion
```

Reply:
Whoops! Well spotted

Reply:
Removed in https://github.com/sky-uk/kfp-operator/pull/384/commits/671ce99fdc84d1fea0ade9f8bac0e6bc44d2ce05